lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -769,7 +825,7 @@ def compare(self, os):
         force_update = False
         if self.endpoint_mode is not None and self.endpoint_mode != os.endpoint_mode:
             differences.add('endpoint_mode', parameter=self.endpoint_mode, active=os.endpoint_mode)
-        if self.env != os.env:
+        if self.env is not None and self.env != os.env:
```

Comment:
`self.env` will never be `None`, it will always be a `dict`

Reply:
Hmm can it not be `None` from ansible parameters?

If no environment is parsed into `env_list` in `get_docker_environment` it will default to None:
https://github.com/hannseman/ansible/blob/68a2543f0bd0d2d44ef16b2575c43760e52aebd5/lib/ansible/modules/cloud/docker/docker_swarm_service.py#L600

Reply:
Ah, sorry, I apparently completely missed the `or None`. Yes, you are totally right!